src/cpufreq.c

```diff
@@ -82,9 +132,67 @@ static int cpufreq_read(void) {
     /* convert kHz to Hz */
     v.gauge *= 1000.0;
 
-    cpufreq_submit(i, v);
+    cpufreq_submit(i, "cpufreq", NULL, &v);
+
+    if (report_p_stats) {
```

Comment:
Please indent error flow; in other words, handle errors. Example:

```c
/* GOOD */
if (!looks_good) {
  handle_error();
  return -1;
}
rest_of_code();
```

instead of:

```c
/* BAD */
if (looks_good) {
  rest_of_code();
} else {
  handle_error();
  return -1;
}
```

In this particular case:

```c
if (!report_p_states) {
  continue;
}
/* rest of code */
```

(this definitely needs to go into the review comments doc)

Reply:
This is not yet addressed.

Reply:
Disagreed, that is not error flow. Imagine we have another `report_other_metrics` variable and another report.... 

If existing code is unacceptable, then we need to move this block of code into own function.
I this will be better solution.

What do you think?

Reply:
Even if it's not error flow: indent the short path, keep the longer / normal path at previous indentation level.

For a more in-depth discussion, see https://github.com/golang/go/wiki/CodeReviewComments#indent-error-flow